File: app/src/main/java/com/yijian/staff/mvp/mine/editpassword/EditPasswordActivity.java
package com.yijian.staff.mvp.mine.editpassword;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

import com.yijian.staff.R;
import com.yijian.staff.widget.NavigationBar;
import com.yijian.staff.widget.NavigationBarItemFactory;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

public class EditPasswordActivity extends AppCompatActivity {

    @BindView(R.id.et_phonenum)
    EditText etPhonenum;
    @BindView(R.id.et_code)
    EditText etCode;
    @BindView(R.id.tv_getcode)
    TextView tvGetcode;
    @BindView(R.id.et_passwd)
    EditText etPasswd;
    @BindView(R.id.et_re_passwd)
    EditText etRePasswd;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_edit_password);
        ButterKnife.bind(this);

        NavigationBar navigationBar = (NavigationBar) findViewById(R.id.edit_password_activity_navigation_bar);
        navigationBar.setTitle("修改密码", "#ffffff");
        navigationBar.setLeftButtonView(NavigationBarItemFactory.createNavigationItemImageView(this, NavigationBarItemFactory.NavigationItemType.BACK_WHITE));
        navigationBar.setLeftButtonClickListener(NavigationBarItemFactory.createBackClickListener(this));


    }

    @OnClick({R.id.tv_getcode, R.id.btn_send})
    public void onViewClicked(View view) {
        switch (view.getId()) {
            case R.id.tv_getcode:
                //TODO 获取验证码

                break;
            case R.id.btn_send:
                //TODO 发送修改密码请求

                break;
        }
    }
}

File: app/src/main/java/com/yijian/staff/mvp/main/mine/editpassword/EditPasswordActivity.kt
package com.yijian.staff.mvp.main.mine.editpassword

import android.os.Bundle
import android.text.TextUtils
import android.widget.EditText
import android.widget.TextView

import com.yijian.staff.R
import com.yijian.staff.mvp.base.mvc.MvcBaseActivity
import com.yijian.staff.net.httpmanager.HttpManager
import com.yijian.staff.net.response.ResultJSONObjectObserver
import com.yijian.staff.prefs.SharePreferenceUtil
import com.yijian.staff.util.CommonUtil

import org.json.JSONObject

import java.util.HashMap

import butterknife.OnClick
import kotlinx.android.synthetic.main.activity_edit_password.*
import kotlinx.android.synthetic.main.view_navigation_bar.*

class EditPasswordActivity : MvcBaseActivity() {



    override fun getLayoutID(): Int {
        return R.layout.activity_edit_password
    }

    override fun initView(savedInstanceState: Bundle?) {

        navigationBar.setTitle("修改密码")
        navigationBar.hideLeftSecondIv()
        navigationBar.setBackClickListener(this)

        val name = SharePreferenceUtil.getUserName()
        if (!TextUtils.isEmpty(name)) {
            tv_user_name.text = name
        }
    }


    @OnClick(R.id.btn_send)
    fun onViewClicked() {
        val username = tv_user_name.text.toString()
        val password = et_password.text.toString()
        if (TextUtils.isEmpty(password)) {
            showToast("原密码不能为空！")
            return
        }
        val psd = et_passwd.text.toString()
        if (TextUtils.isEmpty(psd)) {
            showToast("新密码不能为空！")
            return
        }
        val psd2 = et_re_passwd.text.toString()
        if (TextUtils.isEmpty(psd2)) {
            showToast("确认密码不能为空！")
            return
        }
        if (psd != psd2) {
            showToast("2次输入的密码不同")
            return
        }
        if (!CommonUtil.isPassWordFormat(psd)) {
            showToast("新密码格式不正确,密码是数字和字母的6-20位组合！")
            return
        }
        if (!CommonUtil.isPassWordFormat(psd2)) {
            showToast("确认密码格式不正确,密码是数字和字母的6-20位组合！")
            return
        }

        val params = HashMap<String, String>()
        params["newPwd"] = psd
        params["confirmPwd"] = psd2
        params["originalPwd"] = password
        params["username"] = username
        HttpManager.postHasHeaderHasParam(HttpManager.EDIT_PASSWORD_URL, params, object : ResultJSONObjectObserver(lifecycle) {
            override fun onSuccess(result: JSONObject) {
                setResult(4567)
                finish()
            }

            override fun onFail(msg: String) {
                showToast(msg)
            }
        })
    }
}
